### PR TITLE
Bug/tests

### PR DESCRIPTION
**Pull request by sansive**
## Task
* Add use of IdlingResources to fix the problem related to firebase tasks that take too long
* Add before method on all test classes to dismiss alert dialog in the middle of tests

## To-Do
Remove thread.sleeps on the code for more reliability

### CODE REVIEW

**Review by aindreias — APPROVED** [review]
Nothing much to say for this one. Thanks a lot for your work! Looking forward to ultimately having sleep-free tests 👍

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/TaskListOnMainScreenTest.java [hunk 1]
```diff
@@ -71,6 +75,9 @@ public static void signOut() {
     @Before
     public void forceTaskView() throws InterruptedException {
         Thread.sleep(2000); // no longer necessary to perform clicks
+
+        Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+        context.sendBroadcast(new Intent(Intent.ACTION_CLOSE_SYSTEM_DIALOGS));
```
Comment (Defibrilator) [inline app/src/androidTest/java/com/github/houseorganizer/houseorganizer/TaskListOnMainScreenTest.java:80]: Could extract this to FirebaseTestHelpers

**Review by Defibrilator — CHANGES_REQUESTED** [review]
Nice work thank you ! Just a little extraction and its good

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/TaskListOnMainScreenTest.java (hunk 1 above)
Comment (Defibrilator) [reply app/src/androidTest/java/com/github/houseorganizer/houseorganizer/TaskListOnMainScreenTest.java:80]: ah I see

**Review by Defibrilator — APPROVED** [review]

**Pull request merged**